SQL/proc_descriptions/procedure_compute_speeds_for_segments.md

```diff
@@ -3,29 +3,61 @@
+        - **Else**:
+            - Set `dataset_quality = 1`.
+
+    - **Temporary Table `grouped_speed_records`**:
+        - Structure: `(from_osm_id, to_osm_id, speed, st_dev)`.
+        - If `dataset_quality = 1`, data for this table is sourced from `speed_records`.
+        - If `dataset_quality = 2`, data for this table is sourced from `speed_records_quarterly`.
+
+    - **Insertion into `nodes_ways_speeds`**:
+        - Use `UNION` to merge the selection of data into two groups: ascending sequences and descending sequences.
+        - Insert the merged data into `nodes_ways_speeds`.
+
+This procedure is designed to efficiently calculate and update speed records for given time parameters in a specified target area. The use of temporary tables and indexes ensures optimal performance during the data processing steps.
 
 ## TODO list
 - [x] Create description of the procedure
-- [x] Check if index `target_ways_id_idx` of `target_ways` is used. Is used
-- [x] Check if indexes `node_segments_osm_id_idx`, `node_segments_wf_idx`, `node_segments_wt_idx` of `node_sequences` table are used. Only `node_segments_wf_idx` is used -> Issue.
+- [x] Check if index `target_ways_id_idx` of `target_ways` is used. __Is indeed used__
+- [x] Check if indexes `node_segments_osm_id_idx`, `node_segments_wf_idx`, `node_segments_wt_idx` of `node_sequences` table are used. __Only `node_segments_wf_idx` is used__ -> Issue.
 - [x] Add deeper exaplanation of data selection for `grouped_speed_records` TEMP table. - No need, selection is plain.
-- [x] Replace `..., ...` in explanation of flow with classification name.
-- [ ] Reduction of TEMP tables into CTEs AKA With statements. - `target_ways`, `node_sequences` are candidates, not sure about `grouped_speed_records`.
-- [ ] Reduction of If-else statement with conditional WHERE clause with conditional WHERE clause. - Not easily achived as we have two different source for every case of dataset_quality, and refactored block with great chance would have worse perfomance. Does not worth it, in my opinion.
-- [ ] Reduction of calculation of average speed in 3rd step of main flow - it is calculated twice. May influence the execution perfomance depending on the number of stored data. - Not really sure, cause window function avg() applies after finding records, not before.
+- [ ] Reduction of TEMP tables into CTEs AKA With statements. - `target_ways` - used twice in independent contexts (first one for __STDOUT output__) + additional INDEXes, `node_sequences` - used twice in independent contexts (first one for __STDOUT output__) + 3 INDEXes, `grouped_speed_records` - same as previous two + different table sources are applied to different input of function.
+- [ ] Reduction of If-else statement (1. step) with conditional WHERE clause. - Not easily achived as we have two different source for every case of dataset_quality, and refactored block with great chance would have worse perfomance. Does not worth it, in my opinion.
```

Comment:
I agree, we should only do such a refactoring in case we neet to change the function significantly